templates/.github/workflows/ci.yaml

```diff
@@ -0,0 +1,146 @@
+# This file is synced with beam-community/common-config. Any changes will be overwritten.
+
+name: CI
+
+on:
+  merge_group:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+  push:
+    branches:
+      - main
+  workflow_call:
+    secrets:
+      CI_SERVICE_KEY:
+        required: true
+      GH_PERSONAL_ACCESS_TOKEN:
+        required: true
+      HEX_API_KEY:
+        required: true
+  workflow_dispatch:
+
+concurrency:
+  group: $\{{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  Credo:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Elixir
+        uses: stordco/actions-elixir/setup@v1
+        with:
+          github-token: $\{{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          hex-token: $\{{ secrets.HEX_API_KEY }}
+
+      - name: Credo
+        run: mix credo --strict
+
+  Dependencies:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Elixir
+        uses: stordco/actions-elixir/setup@v1
+        with:
+          github-token: $\{{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          hex-token: $\{{ secrets.HEX_API_KEY }}
+
+      - name: Unused
+        run: mix deps.unlock --check-unused
+
+  Dialyzer:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Elixir
+        uses: stordco/actions-elixir/setup@v1
+        with:
+          github-token: $\{{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          hex-token: $\{{ secrets.HEX_API_KEY }}
+
+      - name: Dialyzer
+        run: mix dialyzer --format github
+
+  Format:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Elixir
+        uses: stordco/actions-elixir/setup@v1
+        with:
+          github-token: $\{{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          hex-token: $\{{ secrets.HEX_API_KEY }}
+
+      - name: Format
+        run: mix format --check-formatted
+
+  Test:
+{{#if IS_MIX_PACKAGE}}
+    name: Test (Elixir $\{{ matrix.versions.elixir }} OTP $\{{ matrix.versions.otp }})
+
+{{/if}}
+    runs-on: ubuntu-latest
+
+    env:
+      MIX_ENV: test
+
+{{#if POSTGRES_VERSION}}
+    services:
+      postgres:
+        image: postgres:{{ POSTGRES_VERSION }}-alpine
+        ports:
+          - 5432:5432
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_INITDB_ARGS: "--nosync"
+
+{{/if}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Elixir
+        uses: stordco/actions-elixir/setup@v1
```

Comment:
I think this action has some stord specific configuration that will need updating